docs: configuration otel headers secretref (#5047)
dapr/docs: daprdocs/content/en/operations/configuration/configuration-overview.md

| `otel.endpointAddress` | string | Set the Open Telemetry (OTEL) server address to send traces to. This may or may not require the https:// or http:// depending on your OTEL provider.
| `otel.isSecure` | bool | Is the connection to the endpoint address encrypted
| `otel.protocol` | string | Set to `http` or `grpc` protocol

> | `otel.headers` | array | Headers to include in OTLP exporter requests. Each entry has a `name` and either a plaintext `value` or a `secretKeyRef` to reference a Kubernetes secret.

Reviewer: This can be any secret, yes, not just a Kubernetes secret. If so, please clarify

Author: Currently this only supports Kubernetes secrets. Supporting other secret stores (like Vault, AWS Secrets Manager, etc.) would be a bigger change - it requires adding an auth.secretStore field to the Configuration CRD. Happy to track that as a follow-up if needed!

Reviewer: Can you add this clarification then. Just say only K8s secrets are supported.

> | `otel.headers` | array | Headers to include in OTLP exporter requests. Each entry has a `name` and either a plaintext `value` or a `secretKeyRef` to reference a Kubernetes secret.

Reviewer: ```suggestion
| `otel.headers` | array | Headers to include in OTLP exporter requests. Each entry has a `name` and either a plaintext `value` or a `secretKeyRef` to reference a Kubernetes secret (only Kubernetes secrets are supported, no other secret type)
```

> | `otel.timeout` | string | Timeout for OTLP exporter requests (e.g. `30s`, `5m`).

Reviewer: ```suggestion
| `otel.timeout` | string | Timeout for OTLP exporter requests (for example `30s`, `5m`).
```

| `OTEL_EXPORTER_OTLP_ENDPOINT` | Sets the Open Telemetry (OTEL) server address, turns on tracing |
| `OTEL_EXPORTER_OTLP_INSECURE` | Sets the connection to the endpoint as unencrypted (true/false) |
| `OTEL_EXPORTER_OTLP_PROTOCOL` | Transport protocol (`grpc`, `http/protobuf`, `http/json`) |
| `OTEL_EXPORTER_OTLP_TRACES_HEADERS` | Comma-separated list of `key=value` headers for the OTLP traces exporter |

> | `OTEL_EXPORTER_OTLP_TRACES_TIMEOUT` | Timeout in milliseconds for the OTLP traces exporter (e.g. `30000`) |

Reviewer: ```suggestion
| `OTEL_EXPORTER_OTLP_TRACES_TIMEOUT` | Timeout in milliseconds for the OTLP traces exporter (for example `30000`) |
```

Is this only in milliseconds?

Author: yes, this followes the OpenTelemetry spec https://opentelemetry.io/docs/specs/otel/configuration/common/#timeout
